tests/coverage.sh

```diff
@@ -39,8 +39,9 @@ RUN find /var/lib -name '*.sh' -exec sed -i 's|#!/bin/sh|#!/bin/bash|g' {} \\; &
     sed -i 's|HOOK_NAME=.*|HOOK_NAME=\${HOOK_NAME:-\$(basename "\$0")}|' /var/lib/githooks/base-template.sh && \\
     sed -i 's|HOOK_FOLDER=.*|HOOK_FOLDER=\${HOOK_FOLDER:-\$(dirname "\$0")}|' /var/lib/githooks/base-template.sh && \\
     sed -i 's|ACCEPT_CHANGES=.*|ACCEPT_CHANGES=\${ACCEPT_CHANGES}|' /var/lib/githooks/base-template.sh && \\
-    sed -i 's|read -r ACCEPT_CHANGES|echo "Accepted: \$ACCEPT_CHANGES" # disabled for tests: read -r ACCEPT_CHANGES|' /var/lib/githooks/base-template.sh && \\
-    sed -i 's|read -r TRUST_ALL_HOOKS|TRUST_ALL_HOOKS=\${TRUST_ALL_HOOKS} # disabled for tests: read -r TRUST_ALL_HOOKS|' /var/lib/githooks/base-template.sh
+    sed -i 's|show_prompt ACCEPT_CHANGES.*|echo "Accepted: \$ACCEPT_CHANGES" # disabled for tests|' /var/lib/githooks/base-template.sh \\
```

Comment:
This is what got us in trouble last time. It mocks the whole prompt logic out, rather than the variable reading bit. I think the change should be (if it works, haven't tested):
```
sed -i 's|read -r "\$VARIABLE"|eval "\$VARIABLE"="\$TEST_ANSWER" # disabled for tests: read -r "\$VARIABLE"|' /var/lib/githooks/base-template.sh && \\
sed -i 's|(call_script "\$DIALOG_TOOL" "githook::" "\$TEXT" "\$SHORT_OPTIONS" "\$@")|(echo "\$TEST_ANSWER") # disabled for tests: call_script "\$DIALOG_TOOL" "githook::" "\$TEXT" "\$SHORT_OPTIONS" "\$@"|' /var/lib/githooks/base-template.sh
```

Reply:
hm... but with your solution we would inject `TEST_ANSWER="asdasd"`. 
But my code just skips the prompt for the **corresponding variable** an replaces it, shouldn't that be fine, its the exact same line where the former `read -r` was...:
```bash
    sed -iE 's|show_prompt ACCEPT_CHANGES.*|echo "Accepted: \$ACCEPT_CHANGES" # disabled for tests|' /var/lib/githooks/base-template.sh \\
    sed -iE 's|show_prompt TRUST_ALL_HOOKS.*|TRUST_ALL_HOOKS=\${TRUST_ALL_HOOKS} # disabled for tests|' /var/lib/githooks/base-template.sh
```

Reply:
The problem here was that the new `read ...` was actually broken, but the test code necer reached there.

Reply:
Aha, true, because of the missing `"$VARIABLE"`.
I think I leave the replacement as I did it, because with your solution, its kind of hard to tell which prompt you mean by injecting the mock `TEST_ANSWER="Y"`... lets see if the tests work...

Reply:
OK, that wasn't working very well, so I tried reusing the previous variables set in the tests already.
It was super painful, but this seems to work well:
```
    sed -i 's|ACCEPT_CHANGES=|ACCEPT_CHANGES=\${ACCEPT_CHANGES}|' /var/lib/githooks/base-template.sh && \\
    sed -i 's%read -r "\$VARIABLE"%eval "\$VARIABLE=\\\\\$\$(eval echo "\\\\\$VARIABLE")" # disabled for tests: read -r "\$VARIABLE"%' /var/lib/githooks/base-template.sh && \\
    sed -i 's%ANSWER=\$(call_script "\$DIALOG_TOOL" "githook::" "\$TEXT" "\$SHORT_OPTIONS" "\$@")%eval "ANSWER=\\\\\$\$(eval echo "\\\\\$VARIABLE")" # disabled for tests: ANSWER=\$(call_script "\$DIALOG_TOOL" "githook::" "\$TEXT" "\$SHORT_OPTIONS" "\$@")%' /var/lib/githooks/base-template.sh
```
Can you please use this?
Also, would you mind duplicating one of the test cases and create a file at `~/".githooks/tools/dialog/run"` so the new code path is also tested?

Thanks!

Reply:
Of course I can use, do you might enligthen me, why my part was not working? You rather like to mocke the show_prompt. right? So your code does that, and mine skips it. ok, fair enough :).
Thanks for the painful support.

Reply:
Yeah, your code was working fine. 👌
It just meant that the whole `show_prompt` function was skipped completely, and the bit that was broken on the first try was in there, so I wanted to see if we can test most of it and just mock out getting the answer.